memdx/errors.go

```diff
@@ -260,3 +261,19 @@ func (e DcpRollbackError) Error() string {
 func (e DcpRollbackError) Unwrap() error {
 	return ErrDcpRollback
 }
+
+type ErrorWithServerDuration struct {
+	Cause          error
+	ServerDuration time.Duration
+}
+
+func (e ErrorWithServerDuration) Error() string {
+	return fmt.Sprintf(
```

Comment:
Unsure we actually want this rather than just passthrough of the message.

Reply:
We do, most other 'enhancement' style error wrappers add their context to the Error string and it makes it a lot easier to debug when you can see what wrappers exist around an error you're printing.